### PR TITLE
CI job for compatibility build + amend code for Boost 1.67

### PR DESCRIPTION
**Pull request by lluiscampos**

### CODE REVIEW

**Review by oleorhagen — APPROVED** [review]

File: .gitlab-ci.yml [hunk 1]
```diff
@@ -42,6 +42,30 @@ test:
     paths:
       - coverage.lcov
 
+backwards-compat:
+  stage: test
+  image: debian:10
```
Comment (kacf) [inline .gitlab-ci.yml:47]: Should put a comment here that this should always be oldest LTS.

**Review by oleorhagen — APPROVED** [review]

File: .gitlab-ci.yml [hunk 2]
```diff
@@ -42,6 +42,30 @@ test:
...
+  before_script:
+    - apt update && apt install -yyq g++ git make lcov pkg-config liblmdb++-dev libboost-dev libboost-log-dev libboost-regex-dev libssl-dev libarchive-dev libdbus-1-dev curl dbus
```
Comment (kacf) [inline .gitlab-ci.yml:49]: Can you add `stunnel4 tinyproxy-bin` right away? Then we won't get a mismerge with #1433.

File: .gitlab-ci.yml [hunk 3]
```diff
@@ -42,6 +42,30 @@ test:
     paths:
       - coverage.lcov
 
+backwards-compat:
+  stage: test
```
Comment (kacf) [inline .gitlab-ci.yml:46]: This is build only, right? Should it be named as such? Or moved to a different stage?

**Review by kacf — APPROVED** [review]

File: .gitlab-ci.yml (hunk 2 above)
Comment (lluiscampos) [reply .gitlab-ci.yml:49]: This job is meant to only build, not to run tests, so it shouldn't be necessary.

Comment (kacf) [reply .gitlab-ci.yml:49]: Doh! Of course. As pointed out by me also ten seconds later! 😄

File: .gitlab-ci.yml (hunk 3 above)
Comment (lluiscampos) [reply .gitlab-ci.yml:46]: I think it belongs to "test" stage in the sense that it tests backwards compatibility. We only want to make sure that it compiles (if the tests cannot run in old LTS we can live with that).

I will name it `test:backward-compat` for consistency with the test stage.

**Pull request merged**